### PR TITLE
Better stripping of strange characters in tagnames when firing metrics.

### PR DESCRIPTION
**Pull request by hodgestar**
Currently we just strip out odd characters but perhaps we should replace them with `_` or `.` if they're not at the start or end of the number?

### CODE REVIEW

**Comment by hodgestar** [comment]
Ready for review.

File: go/vumitools/middleware.py [hunk 1]
```diff
@@ -181,7 +181,8 @@ class MetricsMiddleware(BaseMiddleware):
     """
 
     KNOWN_MODES = frozenset(['active', 'passive'])
-    TAG_SLUGIFIER_RE = re.compile(r"[^a-zA-Z0-9_-]*")
+    TAG_STRIP_RE = re.compile(r"(^[^a-zA-Z0-9_-]+)|([^a-zA-Z0-9_-]+$)")
```
Comment (rudigiesler) [inline go/vumitools/middleware.py:184]: What about `\w-` instead of `a-zA-Z0-9_-`?

Comment (hodgestar) [inline go/vumitools/middleware.py:184]: \w does tricky things with the LOCALE which I was hoping to avoid.

**Comment by rudigiesler** [comment]
:+1: when Travis is happy.

**Pull request closed without merging**